packages/features/audit-logs/lib/handleAuditLogTrigger.ts

```diff
@@ -0,0 +1,209 @@
+import requestIp from "request-ip";
+import type { WebhookDataType } from "webhooks/lib/sendPayload";
+import { z } from "zod";
+
+import { getAuditLogManager } from "@calcom/features/audit-logs/lib/getAuditLogManager";
+import { CRUD, getValues } from "@calcom/features/audit-logs/types";
+import { AuditLogTriggerEvents } from "@calcom/features/audit-logs/types";
+import logger from "@calcom/lib/logger";
+import prisma from "@calcom/prisma";
+import { AppCategories, AuditLogTriggerTargets } from "@calcom/prisma/enums";
+
+const getFirstClause = (userId: number[] | null | undefined, teamId: number | null | undefined) => {
+  const clauses = [];
+  if (userId) {
+    clauses.push({
+      userId: {
+        in: userId,
+      },
+    });
+  }
+
+  if (teamId) {
+    clauses.push({
+      teamId: {
+        equals: teamId,
+      },
+    });
+  }
+
+  if (clauses.length > 1) {
+    return { OR: [...clauses] };
+  } else return clauses[0];
+};
+
+export async function handleAuditLogTrigger({
```

Comment:
this is a work in progress. Comments show what the function is supposed to be in charge of. Its a bit dirty atm and will be done in a couple of days once all eventTriggers have been integrated